### PR TITLE
feat(tui): display git branch

### PR DESCRIPTION
**Pull request by Gustave-241021**
- [x] I have read [`CONTRIBUTING.md`](https://github.com/charmbracelet/.github/blob/main/CONTRIBUTING.md).
- [x] I have created a discussion that was approved by a maintainer (for new features).
    to support : https://github.com/charmbracelet/crush/issues/1596

 Review Summary

  Status: ✅ Approved (with minor suggestions)

  The implementation is solid, following Go idioms and including comprehensive tests.

  Details

  1. New Package: `internal/gitutil`
   * Functionality: Correctly implements directory traversal to find .git and parses the HEAD file.
   * Robustness: Properly handles edge cases like detached HEAD states, non-git directories, and missing files by safely returning an empty string.
   * Testing: branch_test.go provides excellent coverage, including subdirectory traversal and various HEAD file states.

  2. Integration: `header.go` & `splash.go`
   * Logic: The integration cleanly retrieves the branch and conditionally formats the path string.
   * UI/UX: The display logic (Icon + Branch + Separator + Path) is clear and consistent.

  💡 Suggestions

   1. Code Duplication (Formatting):
      The formatting logic for the location string is duplicated in header.go and splash.go:
   1     // In both files
   2     const branchIcon = " "
   3     location = branchIcon + branch + " • " + cwd // (slightly different variable names but same logic)
      Consideration: If you plan to use this formatted string elsewhere, you might move this formatting logic into gitutil or a shared UI helper function (e.g., gitutil.FormatBranchPath(branch, cwd)),
  but for just two occurrences, the current approach is acceptable.

   2. Performance Note:
      GetCurrentBranch performs file I/O (os.Stat and os.ReadFile). While generally fast, ensure this doesn't run on every single render frame (which it doesn't appear to do, as it seems tied to
  specific view updates).

Assisted-by: claude4.5 opus

### CODE REVIEW

**Comment by Gustave-241021** [comment]
image

File: internal/gitutil/branch.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	refPrefix = "ref: refs/heads/"
+)
+
+// GetCurrentBranch returns the current Git branch name for the given directory.
+// It walks up the directory tree to find the .git directory.
+// Returns an empty string if:
+// - The directory is not in a Git repository
+// - The repository is in a detached HEAD state
+// - Any error occurs reading the Git files
+func GetCurrentBranch(dir string) string {
```
Comment (caarlos0) [inline internal/gitutil/branch.go]: its probably simpler to run `git branch --show-current` so we don't really need to handle finding out the `.git` dir 

can also check if its a git repo with `git rev-parse --is-inside-work-tree`

**Comment by meowgorithm** [comment]
Just make sure it has been considered: is there any reason we're not using [go-get](https://github.com/go-git/go-git) here?

**Comment by Gustave-241021** [comment]
Thank you for your feedback!

@caarlos0 i agree, using git CLI directly is much simpler and more reliable, manual parsing is too cumbersome. I will refactor this code when I get home from work. Review it again later pls

@meowgorithm About go git: I think for displaying only branch names, what we need is a lightweight implementation. Importing large libraries like go git may be redundant and unnecessarily increase the size of binary files

**Comment by Gustave-241021** [comment]
I have refactored the relevant code

**Comment by Gustave-241021** [comment]
Is this a change scheduled for the release? 
Maybe it would be cool to release it together with Lazygit.

**Comment by Gustave-241021** [comment]
Btw,should this be a feature within the team plan?

**Review by caarlos0 — APPROVED** [review]
impl looks fine I think

defer to @meowgorithm to see if this is something we want to have.

**Comment by andreynering** [comment]
Hi,

[We recently released a complete UI rewrite](https://github.com/charmbracelet/crush/releases/tag/v0.39.0) to improve performance and maintainability.

The new code is on `internal/ui` as opposed to `internal/tui`, so old PRs will need to be reopened targeting the new codebase.

Thank you!
